src/components/availability-label/helper.ts

```diff
@@ -1,25 +1,81 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import clsx from "clsx";
-import { useGetV1ProductsIdentifier } from "../../core/publizon/publizon";
+import {
+  useGetV1LoanstatusIdentifier,
+  useGetV1ProductsIdentifier
+} from "../../core/publizon/publizon";
 import { useConfig } from "../../core/utils/config";
-import { AccessTypeCode } from "../../core/dbc-gateway/generated/graphql";
+import {
+  Access,
+  AccessTypeCode
+} from "../../core/dbc-gateway/generated/graphql";
 import { FaustId } from "../../core/utils/types/ids";
 import useGetAvailability from "../../core/utils/useGetAvailability";
+import { publizonProductStatuses } from "./types";
 
 export const useAvailabilityData = ({
   accessTypes,
+  access,
   faustIds,
   isbn
 }: {
   accessTypes: AccessTypeCode[];
+  access: Access["__typename"][];
   faustIds: FaustId[] | null;
   isbn: string | null;
 }) => {
   const [isAvailable, setIsAvailable] = useState(false);
   const config = useConfig();
   const isOnline = accessTypes?.includes(AccessTypeCode.Online) ?? false;
+  const [isCostFree, setIsCostFree] = useState<null | boolean>(null);
+  const [isLoading, setIsLoading] = useState<null | boolean>(null);
 
-  useGetAvailability({
+  useEffect(() => {
+    // An online material is by default always available.
+    if (isOnline && isCostFree !== false) {
+      setIsAvailable(true);
+    }
+  }, [isOnline, isCostFree]);
```

Comment:
Maybe I am wrong but the setup here with two requests and useage of `isCostFree` that is used to control `isAvailable` looks fragile and upredictable.

Does the setup work no matter which request resolves first?

Is it not possible to set availability directlty in the `useGetV1ProductsIdentifier()` onSuccess handler?

It would be nice with a unit test that proves that this mechanism works. But I know that it is extra time/scope

Reply:
Let's talk about it together. I have an idea that this could be more robust with a sequential handling of the requests with promises or await.